src/lib.rs

```diff
@@ -1,350 +1,440 @@
+pub fn jaccard_similiarity_score<T>(pred: &[T], actual: &[T]) -> f32
+where
+    T: Eq,
+{
+    return categorical_accuracy(pred, actual);
 }
 
-
 #[cfg(test)]
 mod tests {
     use super::*;
     #[test]
     fn test_gini() {
-        let vec = vec![0, 0, 0, 1];
+        let vec: Vec<usize> = vec![0, 0, 0, 1];
         assert_eq!(0.375, gini(&vec));
-        let v2 = vec![0, 0];
+        let v2: Vec<usize> = vec![0, 0];
         assert_eq!(0.0, gini(&v2));
-        let mut v3 = vec![0];
+        let mut v3: Vec<usize> = vec![0];
         v3.pop();
         assert_eq!(1.0, gini(&v3));
     }
 
     #[test]
     fn test_categorical_accuracy() {
-        let pred = vec![0, 1, 0, 1, 0, 1];
-        let real = vec![0, 0, 0, 0, 1, 0];
+        let pred: Vec<u16> = vec![0, 1, 0, 1, 0, 1];
+        let real: Vec<u16> = vec![0, 0, 0, 0, 1, 0];
         assert_eq!(0.33333334, categorical_accuracy(&pred, &real));
     }
 
     #[test]
     fn test_class_precision() {
-        let actual = vec![0, 1, 2, 0, 1, 2];
-        let pred = vec![0, 2, 1, 0, 0, 1];
-        assert_eq!(0.6666667, class_precision(&pred, &actual, 0));
+        let actual = vec![0_u16, 1, 2, 0, 1, 2];
+        let pred: Vec<u16> = vec![0, 2, 1, 0, 0, 1];
+        assert_eq!(0.6666667, class_precision(&pred, &actual, &0));
     }
 
     #[test]
     fn test_class_recall() {
-        let actual = vec![0, 1, 2, 0, 0, 0];
-        let pred = vec![0, 2, 1, 0, 0, 1];
-        assert_eq!(0.75, class_recall(&pred, &actual, 0));
+        let actual = vec![0_u16, 1, 2, 0, 0, 0];
+        let pred = vec![0_u16, 2, 1, 0, 0, 1];
+        assert_eq!(0.75, class_recall(&pred, &actual, &0));
     }
 
     #[test]
     fn test_weighted_precision() {
-        let actual = vec![0, 1, 2, 0, 1, 2];
-        let pred = vec![0, 2, 1, 0, 0, 1];
+        let actual = vec![0_u16, 1, 2, 0, 1, 2];
+        let pred = vec![0_u16, 2, 1, 0, 0, 1];
         assert_eq!(0.22222224, weighted_precision(&pred, &actual));
     }
 
     #[test]
     fn test_macro_precision() {
-        let actual = vec![0, 1, 2, 0, 1, 2];
-        let pred = vec![0, 2, 1, 0, 0, 1];
+        let actual = vec![0_u16, 1, 2, 0, 1, 2];
+        let pred = vec![0_u16, 2, 1, 0, 0, 1];
         assert_eq!(0.22222222, macro_precision(&pred, &actual));
     }
 
-        #[test]
+    #[test]
     fn test_macro_recall() {
-        let actual = vec![0, 1, 2, 0, 1, 2];
-        let pred = vec![0, 2, 1, 0, 0, 1];
+        let actual = vec![0_u8, 1, 2, 0, 1, 2];
+        let pred = vec![0_u8, 2, 1, 0, 0, 1];
         assert_eq!(0.33333334, macro_recall(&pred, &actual));
     }
 
-        #[test]
+    #[test]
     fn test_weighted_recall() {
-        let actual = vec![0, 1, 2, 0, 1, 2];
-        let pred = vec![0, 2, 1, 0, 0, 1];
+        let actual = vec![0_u8, 1, 2, 0, 1, 2];
+        let pred = vec![0_u8, 2, 1, 0, 0, 1];
         assert_eq!(0.333333334, weighted_recall(&pred, &actual));
     }
 
     #[test]
+    #[cfg(not(any(feature = "use_ndarray")))]
```

Comment:
`#[cfg(not(feature = "..."))]` should be enough. Moreover, the [C-FEATURE](https://rust-lang-nursery.github.io/api-guidelines/naming.html#c-feature) guideline recommends not using placeholder words in Cargo feature names (such as `use`).

Reply:
I will keep this in mind! I've dropped it for now, I'll have a separate branch for feature support. Thanks for the pointer!